server/src/domain/ldap/group.rs

```diff
@@ -154,12 +154,30 @@ fn get_group_attribute_equality_filter(
     is_list: bool,
     value: &str,
 ) -> GroupRequestFilter {
-    deserialize_attribute_value(&[value.to_owned()], typ, is_list)
-        .map(|v| GroupRequestFilter::AttributeEquality(field.clone(), v))
-        .unwrap_or_else(|e| {
+    let value_lc = value.to_ascii_lowercase();
+    let serialized_value = deserialize_attribute_value(&[value.to_owned()], typ, is_list);
+    let serialized_value_lc = deserialize_attribute_value(&[value_lc.to_owned()], typ, is_list);
+    match (serialized_value, serialized_value_lc) {
+        (Ok(v), Ok(v_lc)) => GroupRequestFilter::Or(vec![
+            GroupRequestFilter::AttributeEquality(field.clone(), v),
+            GroupRequestFilter::AttributeEquality(field.clone(), v_lc),
+        ]),
+        (Err(e), Ok(_)) => {
```

Comment:
I think we can still collapse the `(Err(e), _)` case. I just wanted to be explicit about the ok, err

Reply:
Bit unsure what you mean. Remove the `(Err(e), Ok(_))` case, and let it be caught, by making the `(Err(e), Err(_))` into `(Err(e), _)` ?

Reply:
Great, pushed :)